linux_os/guide/system/permissions/files/permissions_within_important_dirs/dir_permissions_library_dirs/oval/shared.xml

```diff
@@ -0,0 +1,46 @@
+<def-group>
+  <definition class="compliance" id="dir_permissions_library_dirs" version="1">
+    {{{ oval_metadata("
+        Checks that /lib, /lib64, /usr/lib, /usr/lib64, /lib/modules, and
+        objects therein, are not group-writable or world-writable.
+      ") }}}
+    <criteria operator="AND">
+      <criterion test_ref="dir_test_perms_lib_dir" />
+      <criterion test_ref="dir_test_perms_lib_files" />
+    </criteria>
+  </definition>
+
+  <unix:file_test check="all" check_existence="none_exist" comment="library directories go-w" id="dir_test_perms_lib_dir" version="1">
+    <unix:object object_ref="dir_object_file_permissions_lib_dir" />
+  </unix:file_test>
+
+  <unix:file_test check="all" check_existence="none_exist" comment="library files go-w" id="dir_test_perms_lib_files" version="1">
+    <unix:object object_ref="dir_object_file_permissions_lib_files" />
+  </unix:file_test>
+
+  <unix:file_object comment="library directories" id="dir_object_file_permissions_lib_dir" version="1">
+    <!-- Check that /lib, /lib64, /usr/lib, /usr/lib64 directories have safe permissions (go-w) -->
+    <unix:path operation="pattern match">^\/lib(|64)|^\/usr\/lib(|64)</unix:path>
+    <unix:filename xsi:nil="true" />
+    <filter action="include">dir_state_perms_nogroupwrite_noworldwrite</filter>
+    <filter action="exclude">dir_perms_state_symlink</filter>
+  </unix:file_object>
+
+  <unix:file_object comment="library files" id="dir_object_file_permissions_lib_files" version="1">
+    <!-- Check the files within /lib, /lib64, /usr/lib, /usr/lib64 directories have safe permissions (go-w) -->
+    <unix:path operation="pattern match">^\/lib(|64)|^\/usr\/lib(|64)</unix:path>
+    <unix:filename operation="pattern match">^.*$</unix:filename>
```

Comment:
```suggestion
    <unix:filename xsi:nil="true" />
```

Reply:
The idea of the rule `dir_object_file_permissions_lib_files` was to match all files in the directories, as far as I understood `  <unix:filename xsi:nil="true" />` will match only the directories, which should have been done in `dir_object_file_permissions_lib_dir`, so my point here is to match all files with `dir_object_file_permissions_lib_files` and dirs and subdirs with `dir_object_file_permissions_lib_dir` object. I guess once you are suggested changes I misunderstood something in the syntax. When I implemented it I had in mind the file_object description in the oval spec (https://oval.mitre.org/language/version5.11/ovaldefinition/documentation/unix-definitions-schema.html), where it is stated that:
```
...
For example, one would set xsi:nil to true if the desire was to test the attributes or permissions associated with a directory. Setting xsi:nil equal to true is different than using a .* pattern match, which says to collect every file under a given path. 
```

Reply:
I have probably misread the comment from brett. If the idea is to check for all files in the directories then it is probably fine. I'm will go through the PR once again soon, please bear with me.

Reply:
Based on text from `SLES-15-010351`, I understand that no files should be checked within this rule, I think that was my original concern.

The library files test should be covered by this rule already:
https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/permissions/files/permissions_within_important_dirs/file_permissions_library_dirs/rule.yml#L43

My question is, do we really want to have the test `dir_object_file_permissions_lib_files` in this rule?

Reply:
@ggbecker  @teacup-on-rockingchair  SLES-15-010352 the STIG for this PR is explicit in covering directories only "-type d", SLES-15-010351 is explicit in covering files "-type f" So, in this case the oval should only be handling directories. So, if I understand this correctly <unix:filename xsi:nil="true" /> is what is needed here.

Reply:
Thanks @brett060102 and @ggbecker , it seems I was totally confused on this one. Dropped completely the rule for files so now(252fa369214604e990f7375532e81448bfeb227e) only dirs are checked by this one.